src/main/java/org/craftercms/studio/model/policy/Action.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.craftercms.studio.model.policy;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Represents a content action that needs to be validated
+ *
+ * @author joseross
+ * @since 3.2.0
+ */
+public class Action {
+
+    /**
+     * The type of action
+     */
+    @NotNull
+    protected Type type;
+
+    /**
+     * The source of the action
+     */
+    protected String source;
+
+    /**
+     * The content-type of the action
+     */
+    protected String contentType;
+
+    /**
+     * The file size of the action
+     */
+    protected long fileSize;
```

Comment:
I think metadata like this property (and content type) might be better to have in a map called `contentMetadata`. In the future, we might need other content properties like mime type, and I think it doesn't make sense to add a bunch of properties here when they are not used all the time and/or depend on the type of content (item vs static asset).